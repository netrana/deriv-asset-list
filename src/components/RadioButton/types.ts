import { RadioChangeEvent } from "antd";

type Option = {
  label: string;
  value: string;
};
export interface Props {
  title?: string;
  options: Option[];
  value?: string;
  className?: string;
  onChange?: (e: RadioChangeEvent) => void;
  defaultValue?: string;
  disabled?: boolean;
}
