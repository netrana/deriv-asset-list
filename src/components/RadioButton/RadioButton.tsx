import React, { FC } from 'react';
import { Radio, Typography } from 'antd';
import classNames from 'classnames';

import { Props } from './types';
import styles from './radioButton.module.scss';

const { Text } = Typography;

const RadioButton: FC<Props> = (props) => {
  const { disabled, title, options, onChange, value, className = '', defaultValue = undefined } = props;

  return (
    <div
      className={classNames({
        [styles.radio]: true,
        [className]: Boolean(classNames),
      })}
    >
      {!!title ? (
        <div className={styles.title}>
          <Text strong>{title}</Text>
        </div>
      ) : null}
      <Radio.Group disabled={disabled} options={options} onChange={onChange} value={value} className={styles.radio} defaultValue={defaultValue} />
    </div>
  );
};

export default RadioButton;
