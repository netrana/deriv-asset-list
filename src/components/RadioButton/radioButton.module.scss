@import 'styles/globals.scss';

.radio {
  :global(.ant-radio-checked .ant-radio-inner) {
    border-color: $primaryThemeColor;
  }
  :global(.ant-radio-checked .ant-radio-inner:after) {
    background-color: $primaryThemeColor;
  }
  .title {
    margin-bottom: 8px;
  }
}
