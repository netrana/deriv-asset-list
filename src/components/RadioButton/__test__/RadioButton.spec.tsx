import React from "react";
import render from "react-test-renderer";

import RadioButton from "../RadioButton";

jest.mock("antd", () => ({
  ...jest.requireActual("antd"),
  Radio: {
    Group: "RadioGroup",
  },
  Typography: {
    Text: "Text",
  },
}));

describe("RadioButton component test", () => {
  describe("snapshot", () => {
    it("should match snapshot without title", () => {
      const tree = render
        .create(
          <RadioButton
            options={[
              {
                value: "1",
                label: "Option 1",
              },
            ]}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should match snapshot with title", () => {
      const tree = render
        .create(
          <RadioButton
            options={[
              {
                value: "1",
                label: "Option 1",
              },
            ]}
            title="Test Title"
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
