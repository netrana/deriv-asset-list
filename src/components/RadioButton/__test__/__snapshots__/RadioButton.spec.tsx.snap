// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component test snapshot should match snapshot with title 1`] = `
<div
  className="radio "
>
  <div
    className="title"
  >
    <Text
      strong={true}
    >
      Test Title
    </Text>
  </div>
  <RadioGroup
    className="radio"
    options={
      Array [
        Object {
          "label": "Option 1",
          "value": "1",
        },
      ]
    }
  />
</div>
`;

exports[`RadioButton component test snapshot should match snapshot without title 1`] = `
<div
  className="radio "
>
  <RadioGroup
    className="radio"
    options={
      Array [
        Object {
          "label": "Option 1",
          "value": "1",
        },
      ]
    }
  />
</div>
`;
